### PR TITLE
updated to tf 1.4 API

### PR DESCRIPTION
**Pull request by kashif**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -833,24 +833,25 @@ distributed training.
 
 Starting in TensorFlow 1.2, there is a new system available for reading data
 into TensorFlow models: dataset iterators, as found in the **tf.contrib.data**
-module. Data iterators are flexible, easy to reason about and to manipulate, and
+module for versions <= 1.3 and **tf.data** module for versions >= 1.4. Data
```
Comment (oahziur) [inline README.md]: We don't really need to mention `tf.contrib.data` here anymore. We have a tf-1.2 branch for tracking 1.3 and 1.2 features.

File: nmt/utils/misc_utils.py [hunk 2]
```diff
@@ -29,7 +29,7 @@
 
 
 def check_tensorflow_version():
-  min_tf_version = "1.4.0-dev"
+  min_tf_version = "1.4.0-rc1"
```
Comment (oahziur) [inline nmt/utils/misc_utils.py]: We would like to keep the `master` tracking tf-nightly, so this should still be `dev`.

I will create a new branch when 1.4.0 is official :)

**Comment by kashif** [comment]
@oahziur should be ready for another review

**Comment by oahziur** [comment]
Thanks! The change has been merged into master branch.

**Pull request closed without merging**